src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -316,14 +317,18 @@ private async Task LoadKeyValuesRegisteredForRefresh(IDictionary<string, Configu
             {
                 string watchedKey = changeWatcher.Key;
                 string watchedLabel = changeWatcher.Label;
+                KeyLabelIdentifier watchedKeyLabel = new KeyLabelIdentifier(watchedKey, watchedLabel);
 
                 // Skip the loading for the key-value in case it has already been loaded
-                if (data.ContainsKey(watchedKey) && data[watchedKey].Label == watchedLabel.NormalizeNull())
+                if (data.ContainsKey(watchedKey) 
+                    && string.Equals(data[watchedKey].Key, watchedKey, StringComparison.Ordinal)

```

Comment:
>data[watchedKey] [](start = 37, length = 16)

The expression `data[watchedKey]` performs hash map lookup. We could do this once and re-use it in other places.